src/doc/man/cargo-yank.md

```diff
@@ -15,16 +15,52 @@ The yank command removes a previously published crate's version from the
 server's index. This command does not delete any data, and the crate will
 still be available for download via the registry's download link.
 
-Note that existing crates locked to a yanked version will still be able to
-download the yanked version to use it. Cargo will, however, not allow any new
-crates to be locked to any yanked version.
+Cargo will not use a yanked version for any new project or checkout without a
+pre-existing lockfile, and will generate an error if there are no longer
+any compatible versions for your crate.
 
 This command requires you to be authenticated with either the `--token` option
 or using {{man "cargo-login" 1}}.
 
 If the crate name is not specified, it will use the package name from the
 current directory.
 
+### How yank works
+
+For example, the `foo` crate published version `0.22.0` and another crate `bar`
+declared a dependency on version `foo = 0.22`. Now `foo` releases a new, but
+not semver compatible, version `0.23.0`, and finds a critical issue with `0.22.0`.
+If `0.22.0` is yanked, no new project or checkout without an existing lockfile will be
+able to use crate `bar` as it relies on `0.22`.
+
+In this case, the maintainers of `foo` should first publish a semver compatible version
+such as `0.22.1` prior to yanking `0.22.0` so that `bar` and all projects that depend
+on `bar` will continue to work.
+
+As another example, consider a crate `bar` with published versions `0.22.0`, `0.22.1`, 
+`0.22.2`, `0.23.0` and `0.24.0`. The following table identifies the versions
+cargo could use in the absence of a lockfile for different SemVer requirements,
+following a given release being yanked:
+
+| Yanked Version / SemVer requirement | `bar = "0.22.0"`                          | `bar = "=0.22.0"` | `bar = "0.23.0"` |
+|-------------------------------------|-------------------------------------------|-------------------|------------------|
+| `0.22.0`                            | Use either `0.22.1` or `0.22.2`           | **Return Error**  | Use `0.23.0`     |
+| `0.22.1`                            | Use either `0.22.0` or `0.22.2`           | Use `0.22.0`      | Use `0.23.0`     |
+| `0.23.0`                            | Use either `0.22.0`, `0.21.0` or `0.22.2` | Use `0.22.0`      | **Return Error** |
+
+### When to yank
+
+Crates should only be yanked in exceptional circumstances, for example,
+license/copyright issues, accidental inclusion of
+[PII](https://en.wikipedia.org/wiki/Personal_data), credentials, etc...
```

Comment:
Nice catch. Updated. 

> Copyright, PII, and credential leakage should email [help@crates.io](mailto:help@crates.io) (or whichever registry you published to) to permanently delete the affected versions.

Do you think it should also mention this? I feel it is a bit much.

Reply:
I think it would be good to mention. I think it helps provide more information on what actions someone should take in those situations (instead of leaving them to guess if they should use `yank` or not).  Another thought is to also maybe point them at https://crates.io/policies? 

I'd also maybe consider removing "credential leakage", and instead say that if you have leaked credentials, the recommended process is to revoke those credentials. Once it is published, it is too late to assume they haven't been copied. I'm not even sure if crates.io will remove a crate in that circumstance (their removal criteria is fairly limited).

Reply:
Updated. Let me know if it is too much :)